Linearize ASG among other improvements (#442)
renuo/applications-setup-guide: README.md

# Renuo Application Setup Guide

This repo is the [Renuo](https://www.renuo.ch) collection of best-practices to set-up apps.
This repo is the [Renuo](https://www.renuo.ch) collection of best-practices to set up apps.
We are a [Rails company](https://rubyonrails.org/foundation), so the most value probably
will be found in the parts concerning Rails. But anyways you'll also find a lot about the
inner workings of Renuo.

* [Ruby on Rails – Application Setup Guide](./ruby_on_rails/README.md)
**You are always welcome to challenge the guide and improve it with a Pull Request.**

## Some Notes on the Side
**Do not blindly follow this guide, always think about what you are doing and why.
If you think something is wrong or simply outdated, improve this guide with a Pull Request.**

If you are reading this document, it means that you have to setup a new application.
A new project started and it's now time to set everything up so that **everyone**,

> in your team, **can start working on it**.

Reviewer: this is also important introduction about what will happen and the main purpose

## How to Use This Guide

This document will try to be as minimalist as possible and provide you with all the steps to set up the application as
fast as possible. There are things, in Renuo projects, which are mandatory, other that are suggested.

> This guide is the result of more than ten years of experience, so this means three things: it's very robust, very opinionated, and possibly very outdated.

Reviewer: this was a very important note to explain how this guide was born and evolved. I'd not drop it.

> 1. **Before You Start** — Read [Naming Conventions](naming_conventions.md), [GitFlow](gitflow.md), and [Security](security.md) to understand the ground rules.

Reviewer: I think you should read the naming convention when you need to give a name to the project. It's of no use at this step, without a purpose. This is covered by the first step in the rails part of the guide.

Knowing giflow is, instead, important, but I don't think it should actually be part of this guide, but of the wiki.

Security is also important but it tells me things I cannot yet do since I don't have a repo, so it should rather be at the very end.

> 3. **Guides & Recipes** — Pick the ones you need (emails, authentication, payment, etc.).

Reviewer: this will slow down the application setup. It's more important to go ahead and setup everything that is mandatory. When the project is ready, everyone can start working and you can go through the optional parts

Before the team can start working on a project, you need:

* A git repository containing the project

> * Two branches: *main* and *develop* (or just *main* for internal projects)

Reviewer: ```suggestion
* Two branches: *main* and *develop* (or just *main* if you decided so)
```

2. **Rails Application Setup** — Follow the [step-by-step setup](ruby_on_rails/README.md) to create, deploy, and configure your app.
3. **Guides & Recipes** — Pick the ones you need (emails, authentication, payment, etc.).
4. **Reference** — Use the [Checklist](checklist.md) to verify you haven't missed anything, and the [Go Live!](go_live.md) page when you're ready to launch.

The basic things that need to be ready before the team can start working on a project are:
## What Needs to Be Ready

* An existing *git* repository containing the project
* Two branches: *main* and *develop*
Before the team can start working on a project, you need:

* A git repository containing the project
* Two branches: *main* and *develop* (or just *main* for internal projects)
* A README with essential information about the application
* Convenience-scripts: `bin/setup`, `bin/check`, `bin/fastcheck`, `bin/run`
* Convenience scripts: `bin/setup`, `bin/check`, `bin/fastcheck`, `bin/run`
* One running, green test
* Continuous integration (*CI*) ready, running and green for both branches
* Continuous deployment (*CD*) ready and running for both branches
* The application deployed for both branches

> As an appendix, you'll find a [checklist](checklist.md) you can use to follow the guide.

Reviewer 2: No more checklist?

**:exclamation: Do not blindly follow this guide, always think about what you are doing and why.
If you think something is wrong or simply outdated, improve this guide with a Pull Request.**

We want you to know exactly the reason behind each single step of this guide.

> Thank you for your work and have fun! :tada:

Reviewer 2: No more fun allowed either xD

> Thank you for your work and have fun! :tada:

Reviewer: yes, this should stay!

> mdbook serve

Reviewer: this does not work (also the previous one did not)

Author: I strongly disagree
```shell
mdbook serve
 INFO Book building has started
 INFO Running the html backend
 INFO HTML book written to `/Users/dani/renuo/projects/application-setup-guide/book`
 INFO Serving on: http://localhost:3000
 ```
